### ISSUE TITLE
As Poppy, I need to log out and return to all my saved information, returning to my roadmap as a landing page

### ISSUE THREAD

**Issue by skeeJay** [newjersey/business.nj.gov#46]
### Completed onboarding scenario
Given I have completed the onboarding form and signed out of the app
When I sign back in later on
Then I am redirected to the roadmap page directly after sign-in
And all of my data is available to edit in the onboarding form

### Brand-new user scenario
Given I am signing in for the first time
When I complete sign-in
Then I am redirected to the landing page directly after sign-in

### Partial onboarding scenario
Given I am signing in after having completed some of the onboarding form (but not all of it)
When I sign-in
Then I am redirected to the landing page directly after sign-in
And all my data so far is pre-populated in the onboarding form.

### on successful auth
- [x] save user isAuth to global authentication context
- [x] make call to backend to get user form data
- [x] if data is 404 or incomplete, send user to onboarding page
- [x] if data is completed, send to roadmap page

### on backend getting user:
- [x] look up data in the database and return it
- [x] return a 404 if user has no data yet

### on logout:
- [x] remove JWT token from localstorage

### on refresh:
- [x] load user details from localstorage 
- [x] set global authentication context for user signed in

### on onboarding form:
- [x] for each page, when they click next, resolve the data with the rest of user's data and send a post and mutate local user data

### on backend posting user data:
- [x] if user exists, update them
- [x] if user does not exist, create them

**Comment by aloverso** [comment]
@skeeJay that's fair. We should discuss this. Some options:
- the landing page should redirect you to Roadmap immediately if you are signed in
- the landing page should look & feel different for signed-in users in general, and should have a call-to-action that takes them to their existing roadmap instead of the onboarding flow

**Comment by skeeJay** [comment]
@aloverso My default inclination is the first option, mainly because I think if you're returning without logging out you've been here recently, you're maybe returning often, and that feels like a repeat/power user. Also, we intended the roadmap to be that kind of homepage/home base, so I think as a paradigm it makes sense. If the roadmap page isn't providing enough context for a return/repeat logged in user landing right on it, I think maybe that is something to discuss about the roadmap page before we start adding "double duty" to the introduction/landing page.

**Comment by aloverso** [comment]
That's reasonable. Can we make that a separate story? Going to the "root" url should automatically redirect you to your roadmap page if you're already signed in

**Comment by skeeJay** [comment]
Created as #65. Since all the other criteria of this ticket seems to be met, marking as approved.
